editor/src/components/PropertiesPanel/PropertiesPanelDynamic.vue

```diff
@@ -0,0 +1,211 @@
+import { getVariableName } from '../../model/uischema';
+import { tryFindByUUID } from '../../util/schemasUtil';
+import _ from 'lodash';
+const PropertiesPanel = defineComponent({
+  name: 'PropertiesPanel',
+  props: {},
+  components: {
+    PropertiesPanelDynamic,
+  },
+  data() {
+    return {
+      panel: [0, 1, 2, 3, 4],
+      generalData: null,
+      type: null,
+      uiElement: null,
+    };
+  },
+  computed: {
+    schema: sync('app/editor@schema'),
+    uischema: sync('app/editor@uiSchema'),
+    selectedElement: sync('app/editor@element'),
+  },
+  mounted() {
+    this.setSelection(this.selectedElement.selected);
+  },
+  watch: {
+    selectedElement(newSelection, oldSelection) {
+      this.setSelection(newSelection.selected);
+    },
+  },
+  methods: {
+    setSelection: function (newSelection) {
+      let fieldData = {};
+      let linkedSchemaUUID, elementSchema;
+      // UIELEMENT is the ui schema for specific scope
+      this.uiElement = tryFindByUUID(this.uischema, newSelection);
+
+      if (this.uiElement) {
+        linkedSchemaUUID = this.uiElement.linkedSchemaElement;
+        // ELEMENTSCHEMA is the schema for specific scope
+        elementSchema =
+          linkedSchemaUUID && this.schema
+            ? tryFindByUUID(this.schema, linkedSchemaUUID)
+            : undefined;
+      }
+
+      fieldData = omit(this.uiElement, [
+        'uuid',
+        'parent',
+        'elements',
+        'linkedSchemaElement',
+        'options.detail',
+        'scope',
+        'type',
+      ]);
+      fieldData['variable'] = getVariableName(this.uiElement);
+      if (elementSchema && elementSchema.schema) {
+        // Get the required property
+        fieldData['required'] =
+          this.schema.schema.required &&
+          this.schema.schema.required.includes(getVariableName(this.uiElement));
+        // Get the description property
+        fieldData['description'] = elementSchema.schema.description
+          ? elementSchema.schema.description
+          : '';
+        // Get the maLength property
+        fieldData['maxLength'] = elementSchema.schema.maxLength
+          ? elementSchema.schema.maxLength
+          : '';
+
+        fieldData['readOnly'] = elementSchema.schema.readOnly
+          ? elementSchema.schema.readOnly
+          : false;
+      }
+      this.generalData = {
+        type: this.uiElement.type,
+        data: fieldData,
+      };
+    },
+    updateData(data: any) {
+      const elementSchema = this.findElementSchema();
+      this.generalData['data'] = data;
+      // variable
+      if (data.variable) {
+        this.$store.dispatch('app/updateSchemaVariable', {
+          elementUUID: this.uiElement.uuid,
+          newVariable: data.variable,
+        });
+        this.$store.dispatch('locales/updateProperty', {
+          oldProperty: elementSchema ? elementSchema.key : '',
+          newProperty: data.variable,
+        });
+      }
+      // required
+      if (data.required) {
```

Comment:
please create a ticket to improve this  code section.

Reply:
https://github.com/rodriquelca/dunamis-jform/issues/24
Issue created